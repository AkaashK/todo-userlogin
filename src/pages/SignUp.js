import React, { useState } from "react";
import logo from "../logo.svg";
import "../App.css";
import History from "./history";
import store from "./store/store";
import AddUser from "./store/actions";

function SignUp() {
  const [user, setUser] = useState({ name: "", email: "", password: "" });

  const { email, password, name } = user;

  function signup() {
    if (name === "" || email === "" || password === "") {
      document.getElementById("alert").innerHTML = "please enter all fields";
    } else {
      store.dispatch(AddUser({ name, email, password }));
      History.push("/login");
    }
  }

  const handleChange = (name) => (event) => {
    document.getElementById("alert").innerHTML = "";
    setUser({ ...user, [name]: event.target.value });
  };

  return (
    <div className="App-header">
      <img src={logo} className="App-logo" alt=""></img>
      <h2 className="title">SIGN UP</h2>
      <p id="alert" data-testid="alertmsg"></p>
      Name
      <input
        id="name"
        data-testid="nameTest"
        type="text"
        value={name}
        onChange={handleChange("name")}
        required
      />{" "}
      <br />
      email
      <input
        id="email"
        data-testid="emailTest"
        type="text"
        value={email}
        onChange={handleChange("email")}
        required
      />{" "}
      <br />
      password
      <input
        id="password"
        data-testid="passTest"
        type="password"
        value={password}
        onChange={handleChange("password")}
        required
      />{" "}
      <br />
      <button
        data-testid="button"
        id="loginbtn"
        className="btn"
        type="button"
        onClick={signup}
      >
        signup
      </button>
    </div>
  );
}

export default SignUp;
