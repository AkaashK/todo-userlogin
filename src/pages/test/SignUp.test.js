import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import SignUp from "../SignUp";
import "@testing-library/jest-dom/extend-expect";

const renderComponent = () => render(<SignUp />);

afterEach(cleanup);
describe("SignUp", () => {
  it("renders correctly", () => {
    const component = renderComponent()
    expect(component).toMatchSnapshot();
  });

  it("shows button text", () => {
    const { getByTestId } = renderComponent();
    expect(getByTestId("button").textContent).toBe("signup");
  });

  it("should get user value", () => {
    const { getByTestId } = renderComponent();
    const emailid = getByTestId("emailTest");
    const passid = getByTestId("passTest");
    fireEvent.change(emailid, { value: "hello" });
    fireEvent.change(passid, { value: "123123" });
    expect(getByTestId("alertmsg").textContent).toBe("");
    // expect(getByTestId('emailTest').textContent).toBe('hello')
    // expect(getByTestId)
  });

  it("should get user alert", () => {
    const { getByTestId } = renderComponent();
    const emailid = getByTestId("emailTest");
    const passid = getByTestId("passTest");
    const id = getByTestId("button");
    fireEvent.click(id);
    fireEvent.change(emailid, { value: "" });
    fireEvent.change(passid, { value: "" });
    expect(getByTestId("alertmsg").textContent).toBe("please enter all fields");
  });

  it("should get user alert with any one value", () => {
    const { getByTestId } = renderComponent();
    const emailid = getByTestId("emailTest");
    const passid = getByTestId("passTest");
    const id = getByTestId("button");
    fireEvent.click(id);
    fireEvent.change(emailid, { value: "hello" });
    fireEvent.change(passid, { value: "" });
    expect(getByTestId("alertmsg").textContent).toBe("please enter all fields");
  });
});
