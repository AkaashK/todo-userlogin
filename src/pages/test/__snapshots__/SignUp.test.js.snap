// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App-header"
      >
        <img
          alt=""
          class="App-logo"
          src="logo.svg"
        />
        <h2
          class="title"
        >
          SIGN UP
        </h2>
        <p
          data-testid="alertmsg"
          id="alert"
        />
        Name
        <input
          data-testid="nameTest"
          id="name"
          required=""
          type="text"
          value=""
        />
         
        <br />
        email
        <input
          data-testid="emailTest"
          id="email"
          required=""
          type="text"
          value=""
        />
         
        <br />
        password
        <input
          data-testid="passTest"
          id="password"
          required=""
          type="password"
          value=""
        />
         
        <br />
        <button
          class="btn"
          data-testid="button"
          id="loginbtn"
          type="button"
        >
          signup
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App-header"
    >
      <img
        alt=""
        class="App-logo"
        src="logo.svg"
      />
      <h2
        class="title"
      >
        SIGN UP
      </h2>
      <p
        data-testid="alertmsg"
        id="alert"
      />
      Name
      <input
        data-testid="nameTest"
        id="name"
        required=""
        type="text"
        value=""
      />
       
      <br />
      email
      <input
        data-testid="emailTest"
        id="email"
        required=""
        type="text"
        value=""
      />
       
      <br />
      password
      <input
        data-testid="passTest"
        id="password"
        required=""
        type="password"
        value=""
      />
       
      <br />
      <button
        class="btn"
        data-testid="button"
        id="loginbtn"
        type="button"
      >
        signup
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
